### PR TITLE
fix gateway server port analysis result of GatewayPortNotOnWorkload is incorrect

### PR DESCRIPTION
**Pull request by nicole-lihui**
**Please provide a description of this PR:**

The Analyzer for GatewayPortNotOnWorkload was producing incorrect results due to associating the gateway.servers[].port (the port for ingress listeners) with the `Port` of the Service.

The correct association should be with the `TargetPort` of the Service.

### CODE REVIEW

File: pkg/config/analysis/analyzers/gateway/gateway.go
Comment (hanxiaop) [inline pkg/config/analysis/analyzers/gateway/gateway.go:87]: Should we update the descriptions above as well?

Comment (hanxiaop) [reply pkg/config/analysis/analyzers/gateway/gateway.go:87]: https://github.com/istio/istio/blob/f336e74304772bcc91100d3c7775524b2b18448a/pkg/config/analysis/analyzers/gateway/gateway.go#L32-L49

some comments and also the analyzer description

Comment (nicole-lihui) [reply pkg/config/analysis/analyzers/gateway/gateway.go:89]: I believe it is necessary to provide a comment explaining why the choice of targetPort is made, as I have also had doubts about it in the past.

**Review by hanxiaop — APPROVED** [review]
lgtm, thanks

File: pkg/config/analysis/analyzers/gateway/gateway.go
Comment (hanxiaop) [inline pkg/config/analysis/analyzers/gateway/gateway.go:89]: I'm actually a little bit confused about the message description `The gateway refers to a port that is not exposed on the workload (pod selector %s; port %d)`, since the gateway is specifying which port to listen on.

Comment (hanxiaop) [reply pkg/config/analysis/analyzers/gateway/gateway.go:89]: Not blocking this PR, and I think we may need to rephrase the message description as a followup.

**Comment by nicole-lihui** [comment]
@istio/wg-networking-maintainers  pls review, thanks

**Review by ramaraochavali — APPROVED** [review]

**Comment by ericvn** [comment]
It looks like this PR likely has broken the istio.io test. See https://storage.googleapis.com/istio-prow/pr-logs/pull/istio_istio.io/13526/doc.test.profile-none_istio.io/1678493306809487360/artifacts/tests-setup-profile-none-c4cfae/TestDocs/setup/getting-started/test.sh/test.sh/_test_context/test.sh_output.txt and the getting started test.

**Comment by ericvn** [comment]
I suspect it's the verify here that's failing: https://preliminary.istio.io/latest/docs/setup/getting-started/#ip

**Comment by hanxiaop** [comment]
Yes I think this results in the test failure. I tried the analyze and it gives me the same warning.

**Comment by hanxiaop** [comment]
I'm looking into it

**Pull request merged**
